services/web/client/source/class/osparc/ui/table/cellrenderer/Percentage.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * oSPARC - The SIMCORE frontend - https://osparc.io
+ * Copyright: 2019 IT'IS Foundation - https://itis.swiss
+ * License: MIT - https://opensource.org/licenses/MIT
+ * Authors: Ignacio Pascual (ignapas)
+ */
+
+qx.Class.define("osparc.ui.table.cellrenderer.Percentage", {
+  extend: qx.ui.table.cellrenderer.Html,
+
+  construct: function(color) {
+    this.base(arguments, "center");
+    this.setColor(color);
+  },
+
+  properties: {
+    color: {
+      check: "String",
+      nullable: false
+    },
+    unit: {
+      check: "String",
+      nullable: false,
+      init: "%"
+    }
+  },
+
+  members: {
+    // overridden
+    _getContentHtml: function(cellInfo) {
+      if (cellInfo.value == null || cellInfo.value < 0) { // eslint-disable-line no-eq-null
+        return "";
+      }
+      const splitted = cellInfo.value.split("/");
+      const width = typeof parseFloat(splitted[0]) === "number" && splitted.length === 2 ? this._calculateWidthPercentage(splitted[0], splitted[1]) : 0;
+      return "" +
+        `<div style="position: absolute; left: 0; right: 0;">${splitted[0]} ${this.getUnit()}</div>` +
```

Comment:
I am not 100% percent sure of what you mean... but I think it would be too complicated and problem-prone, if ever possible.
Modern web frameworks are not bare JS libraries you can import and use. They come with a full set of tools, workflows and syntaxes. They also need its own transpilers and file structures...
For example, React is not just the components with internal state that re-render whenever it changes, and the life-cycle functions.
It is also the awesome JSX language/syntax, that allows you to structure your components and views in a html-like fashion. It is the router, that handles the different views of the application, the links between them and synchronizes with the browser history and url bar.
It is redux, that integrates a centralized data store (state) of the application within every UI component, that automagically update reflecting the changes.
It is flux, a design pattern that works very well with redux, allowing to replicate any past state of the interface, and thus allowing for ctrl+z undo function to be easily implemented.
If I was using several frameworks in the application, which would increase the size of the application bundle (amount of JS code sent to the client, already around 12mb) and I don't recommend, I would put the more advanced/flexible ones on top.
I am sorry I cannot explain all this better...

Reply:
I think I know what you mean @pcrespov . You say we could write the components in a similar fashion and do the styling in the css files. I guess that is possible (although it goes against the motivations behind Qooxdoo), but it has some maintenance and scalability problems. You have to style your components so they match the style of those from Qooxdoo, styled by the theme. Now you have the styles coming from different sources. Now if you want to change the theme, or it updates, you have to go through all the css to match these changes. It would be doable though... just not very cool